src/Model/Behavior/UploadBehavior.php

```diff
@@ -55,7 +56,7 @@ public function initialize(array $config): void
         $schema = $this->_table->getSchema();
         /** @var string $field */
         foreach (array_keys($this->getConfig()) as $field) {
-            $schema->setColumnType($field, 'upload.file');
+            $schema->addColumn($field, ['type' => 'upload.file']);
```

Comment:
A much simpler solution is just wrapping the `setColumnType()` call within a `hasColumn()`, which won't require any additional changes.

Reply:
Just saw your suggestion, I'll try that

Reply:
I fact, I don't think so. Those fields are "virtual". They don't reside in the table. hasColum will always return false

Reply:
Yes, it would and hence nothing needs to be done. In 5.0 the `setColumnType()` call simply returned without doing anything if the column did not actually exist in the schema. So things remain the same.

Reply:
In 5.0, it would add the non-existant column in the schema, or I did something wrong

```
object(Cake\Database\Schema\TableSchema) id:0 {
'table' => 'attachments'
'columns' => [
    'id' => [ ],
    'name' => [ ],
    'size' => [ ],
    'type' => [ ],
    'dir' => [ ],
    'created' => [ ],
    'modified' => [ ],
    'email_heading' => [ ],
    'thumbnail' => [ ],
    'poster' => [ ],
    'illustration' => [ ],
]
```

Reply:
> In 5.0, it would add the non-existant column in the schema...

No it doesn't https://github.com/cakephp/cakephp/blob/5.0.11/src/Database/Schema/TableSchema.php#L379-L383

Reply:
ah yes, sorry, I didn't clear the cache after having downgraded to 5.0.